docs(utils): remove TabFocusTrap and add docs for utils (#1577)
contentful/forma-36: packages/components/utils/src/Portal/Portal.mdx

> ---
> title: 'Portal'
> type: 'component'
> status: 'beta'
> slug: /utils/portal/
> typescript: ./Portal.tsx
> ---

Reviewer: I notice that you are already using the new format of the documentation in the body of this page
and that's great!

But I think I merged some changes to the plop template while this PR is in review and now I think it would be a good idea to rebase and implement some changes to the new pages

the first one is that in the frontmatter of these pages we need to pass `v4Doc: true` and we still didn't designed where to show the status badge so I think it can be removed for now. This is how the frontmatter will look like:

```suggestion
---
title: 'Portal'
type: 'component'
slug: /utils/portal/
typescript: ./Portal.tsx
v4Doc: true
---
```

Another change is the "Props (API reference)" title
I will add the comment with the changes in that line (to be continued)

Author: ok, no problem

import { Props } from '@contentful/f36-docs-utils';

`Portal` component can be used to render children in a DOM node outside of its parent component. It can be useful for rendering components that you don’t want to be clipped by their parent’s component. For example: modals, popovers, dropdown, etc.

## Table of contents

- [Import](#import)
- [Examples](#examples)
  - [Basic usage](#basic-usage)
- [Props](#props)

## Import

```jsx static=true
import { Portal } from '@contentful/f36-utils';
```

## Example

### Basic usage

```jsx static=true
<Flex>
  <Portal>
    <Flex
      style={{
        backgroundColor: 'gray',
      }}
    >
      Hi! I'm first in the document flow, but will be rendered at the end of the
      document!
    </Flex>
  </Portal>
  Parent component content
</Flex>
```

> ## Props

Reviewer: We now have a "Open in Storybook" button aligned with this title
and because of that we need to use some react components here

the [Accordion doc](https://github.com/contentful/forma-36/blob/forma-v4/packages/components/accordion/Accordion.mdx) has a good example of what needs to be done, but for the Portal page the code looks like this:

```suggestion
<Stack justifyContent="space-between" marginTop="spacing2Xl" marginBottom="spacingL">
  <Heading id="props-api-reference" as="h2" marginTop="none" marginBottom="none">
    Props (API reference)
  </Heading>

  <TextLink
    href="https://v4-f36-storybook.netlify.app/?path=/story/utilities-portal"
    target="_blank"
    alignIcon="end"
    icon={<ExternalLinkIcon />}
  >
    Open in Storybook
  </TextLink>
</Stack>
```

As you can see, it's necessary to import Stack, Heading, and TextLink on the top of the MDX file or this will fail

Author: but if there is no storybook page? should I add it?

Reviewer: Good point!

I guess in that case, you can just do `## Props (API reference)` if they have any props

> ## Props (API reference)

Reviewer: I think Portal has a story in storybook, no?

Isn't this the same Portal that this doc is about?
https://v4-f36-storybook.netlify.app/?path=/story/utilities-portal

Author: yes, you are right!

---
title: 'Portal'
type: 'component'
status: 'beta'
slug: /utils/portal/
typescript: ./Portal.tsx
---

import { Props } from '@contentful/f36-docs-utils';

> `Portal` component can be used to render children in to a DOM node outside of the parent component. It can be useful for rendering components, you dont want to be hidden, clipped by parent component. For example, models, popovers, dropdowns.

Reviewer: ```suggestion
`Portal` component can be used to render children in a DOM node outside of its parent component. It can be useful for rendering components that you don’t want to be clipped by their parent’s component. For example: modals, popovers, dropdown, etc.
```

## Table of contents

- [Import](#import)
- [Examples](#examples)
  - [Basic usage](#basic-usage)
- [Props](#props)
- [Help improve this page](#help-improve-this-page)

## Import

```jsx static=true
import { Portal } from '@contentful/f36-utils';
```

## Example

### Basic usage

```jsx static=true
<Box>
  <Portal>
    <Box
      style={{
        backgroundColor: 'gray',
      }}
    >
      Hi! I'm first in the document flow, but will be rendered at the end of the
      document!
    </Box>
  </Portal>
  Parent component content

> </Box>

Reviewer: I don't understand the Box component here

Shouldn't it be a Fragment or a Flex?
